src/main/java/trackie/storage/TaskList.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    public void addDeadlineTask(String[] arguments) throws TrackieException {
+        if (arguments.length == 1) {
+            throw new TrackieException("Correct usage: deadline [desc] /by [yyyy-mm-dd]");
+        }
+
+        String desc;
+        String deadline;
+
+        //retrieve the description
+        int ptr = 1;
+        StringBuilder sb = new StringBuilder();
+        StringBuilder sb2 = new StringBuilder();
+
+        while (!arguments[ptr].equals("/by")) {
+            if (ptr == arguments.length - 1) {
+                throw new TrackieException("Correct usage: deadline [desc] /by [yyyy-mm-dd]");
+            }
+            sb.append(arguments[ptr]).append(' ');
+            ptr++;
+        }
+        if (sb.isEmpty()) {
+            throw new TrackieException("Description cannot be empty!");
+        } else {
+            desc = sb.substring(0, sb.length() - 1);
+        }
+
+        //retrieve the deadline
+        ptr++;
+        while (ptr < arguments.length) {
+            sb2.append(arguments[ptr]).append(' ');
+            ptr++;
+        }
+
+        if (sb2.isEmpty()) {
+            throw new TrackieException("Deadline cannot be empty!");
+        } else {
+            deadline = sb2.substring(0, sb2.length() - 1);
+        }
+
+        try {
+            Task instance = new Deadline(desc, deadline);
+            tasks.add(instance);
+
+            System.out.printf("Added: [D][%s] %s\n", instance.getStatusIcon(), instance.getTaskInfo());
+            System.out.printf("You now have %d task(s) in total.\n", tasks.size());
+        } catch (DateTimeParseException e) {
+            System.out.println("Please use the correct format: yyyy-mm-dd");
+            System.out.println("(Also check that you are entering a valid date!)");
+        }
+
+    }
+
+    public void addEventTask(String[] arguments) throws TrackieException {
+        if (arguments.length == 1) {
+            throw new TrackieException("Correct usage: event [desc] /from [start] /to [end]");
+        }
+
+        String desc = "";
+        String start = "";
+        String end = "";
+
+        //retrieve the description
+        int ptr = 1;
+        StringBuilder sb = new StringBuilder();
+        while (!arguments[ptr].equals("/from")) {
+            if (ptr == arguments.length - 1) {
+                throw new TrackieException("Correct usage: event [desc] /from [start] /to [end]");
+            }
+            sb.append(arguments[ptr]).append(' ');
+            ptr++;
+        }
+        if (sb.isEmpty()) {
+            throw new TrackieException("Description cannot be empty!");
+        } else {
+            desc = sb.substring(0, sb.length() - 1);
+        }
+
+        ptr++;
+        if (ptr >= arguments.length) {
+            throw new TrackieException("Correct usage: event [desc] /from [start] /to [end]");
+        }
+        //retrieve the start time
+        StringBuilder sb2 = new StringBuilder();
```

Comment:
Perhaps you could use another way of naming to distinguish the StringBuilders rather than using numbers.